packages/react-pose-core/src/components/Transition/index.tsx

```diff
@@ -23,12 +23,14 @@ class Transition extends React.PureComponent<Props> {
 
   removeChild(key: string) {
     const { children, leaving } = this.state;
-    const { enterAfterExit } = this.props;
+    const { enterAfterExit, onRest } = this.props;
     if (!leaving.hasOwnProperty(key)) return;
 
     leaving[key] = true;
```

Comment:
does this one has to be kept in state? its part of the state and u mutate it here, which is imho surprising in react's world, so maybe a class instance field would suffice for this? i generally try to stick to keeping in react's state only things related to render

Reply:
`getDerivedStateFromProps` is a static field so the only way to provide it scoped state is via `state`.

Mutating `leaving` is a bit naughty by the book (and we also provide a scoped function in state for the same reason that never gets updated) but if it's not done ignorant of the purpose of `state` and its immutability, there's nothing technically wrong with it.

Reply:
Fair point about `leaving` having to be available in `getDerivedStateFromProps`. Maybe add a note/warning about leaving being part of the state but still being mutated here? Or don't destructure it out of `this.state` to make it more visible. 

Either way - logic is ok, so if you don't think a change is necessary then don't change anything :)